### ISSUE TITLE
Rule introduced in #1454 matches against other kubernetes resources containing secret in their kind field.

### ISSUE THREAD

**Issue by marcm-ml** [gitleaks/gitleaks#1508]
I think the rule introduced in #1454 was not implemented in consideration of other custom resources that might match `\bkind:.{0,10}Secret\b` and sops-encrypted files.

For example, I get plenty of false-positive for resource definitions `kind: ExternalSecret` as well as sops-encrypted files for `kind: SopsSecret`. Similarly, using FluxCD's feature of decrypting sops-encrypted files for normal "kind: Secret" is also matching as false-positives.

Since the "data:" keyword appears in both files regularly, it will match the regex of the rule. If I modify the rule to only match against "kind: Secret" all findings matching different kinds go away. This is my config:

```toml
[extend]
useDefault = true

[[rules]]
id = "kubernetes-secret-with-data-after"
regex = '''(?i)\b((?i)(?:\bkind:[\"']?Secret[\"']?\b)(?:.|\s){0,200}?\b(?:data:)\s*(.+))(?:['|\"|\n|\r|\s|\x60|;]|$)'''

[[rules]]
id = "kubernetes-secret-with-data-before"
regex = '''(?i)\b((?i)(?:\b(?:data:))(\W+(?:\w+\W+){0,200}?)\bkind:[\"']?Secret[\"']?\b)(?:['|\"|\n|\r|\s|\x60|;]|$)'''
```

However, I still get findings for sops-encrypted which are normal k8s secrets. For that I would have to edit the regex to check if a "sops" metadata field is present. I am not sure if this is sufficient but its atleast working for me and would still detect secrets for "kind: Secret"

```toml
[extend]
useDefault = true

[[rules]]
id = "kubernetes-secret-with-data-after"
regex = '''(?i)\b((?i)(?:\bkind:[\"'\s]*Secret[\"']?\b)(?:.|\s){0,200}?\b(?:data:)\s*(.+)\b(?:sops:){0})(?:['|\"|\n|\r|\s|\x60|;]|$)'''

[[rules]]
id = "kubernetes-secret-with-data-before"
regex = '''(?i)\b((?i)(?:\b(?:data:))(\W+(?:\w+\W+){0,200}?)\bkind:[\"'\s]*Secret[\"']?\b)\b(?:sops:){0}(?:['|\"|\n|\r|\s|\x60|;]|$)'''
```

I still get false-positives if the yaml file is a multi-document and contains both "kind: Secret" and "kind: ExternalSecret".

References:

https://github.com/isindir/sops-secrets-operator
https://external-secrets.io/latest/api/externalsecret/
https://fluxcd.io/flux/guides/mozilla-sops/

**Comment by rgmz** [comment]
Good to know. I'll incorporate this into #1462.

**Comment by fabio-sv** [comment]
Facing the same issue with `ExternalSecret` resources.
A potential fix is to extend the regex to allow the secret if the string `remoteRef` is supplied after `data:`

**Comment by rgmz** [comment]
> A potential fix is to extend the regex to allow the secret if the string remoteRef is supplied after data:

@fabio-sv do you have an example of this that I can add to the test cases for #1462?

**Comment by fabio-sv** [comment]
@rgmz Sure, here's an example resource:

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: api-key
  namespace: my-namespace
spec:
  secretStoreRef:
    kind: ClusterSecretStore
    name: aws-secretsmanager
  refreshInterval: 1h
  target:
    creationPolicy: Owner
  data:
    - secretKey: api-key
      remoteRef:
        key: my-secrets-manager-secret
```

**Comment by rgmz** [comment]
This should be fixed by #1462.

**Comment by marcm-ml** [comment]
Closing as fixed for me
